smoderp2d/time_step.py

```diff
@@ -1,265 +1,354 @@
 # @package smoderp2d.time_step methods to perform
 #  time step, and to store intermediate variables
 
+from itertools import cycle
+from uu import Error
+from matplotlib.pylab import norm
 from smoderp2d.core.general import Globals, GridGlobals
 import smoderp2d.processes.rainfall as rain_f
-import smoderp2d.processes.infiltration as infilt
+import smoderp2d.processes.infiltration as infiltration
```

Comment:
Although I do agree with this change in general, I do not think it should be part of this PR but rather of a separate one. It just adds changes unrelated to the implicit solution.